includes/stats-calculator.php

```diff
@@ -253,4 +255,40 @@ public function event_has_stats( int $event_id ): bool {
 
 		return ! empty( $stats->rows() );
 	}
+
+	/**
+	 * Check if a user is a first time contributor.
+	 *
+	 * @param Event_Start_Date $event_start The event start date.
+	 * @param int              $user_id      The user ID.
+	 *
+	 * @return bool True if the user is a first time contributor, false otherwise.
+	 */
+	public function is_first_time_contributor( $event_start, $user_id ) {
+		global $wpdb, $gp_table_prefix;
+
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		// phpcs:disable WordPress.DB.DirectDatabaseQuery.DirectQuery
+		// phpcs:disable WordPress.DB.DirectDatabaseQuery.NoCaching
+		// phpcs:disable WordPress.DB.DirectDatabaseQuery.SchemaChange
+		$users_first_translation_date = $wpdb->get_var(
+			$wpdb->prepare(
+				"
+			select min(date_added) from {$gp_table_prefix}translations where user_id = %d
+		",
+				array(
+					$user_id,
+				)
+			)
+		);
+
+		if ( get_userdata( $user_id ) && ! $users_first_translation_date ) {
+			return true;
+		}
+		$event_start_date_time  = new DateTimeImmutable( $event_start->__toString(), new DateTimeZone( 'UTC' ) );
+		$first_translation_date = new DateTimeImmutable( $users_first_translation_date, new DateTimeZone( 'UTC' ) );
+		$multiplier             = $users_first_translation_date > $event_start ? 1 : -1;
+		$date_diff              = ( $event_start_date_time->diff( $first_translation_date )->days ) * $multiplier;
+		return $date_diff >= -1;
```

Comment:
You can compare `DateTimeImmutable` objects in PHP using standard comparison operators, so I think this code is simpler:

```suggestion
		if ( $event_start_date_time <= $first_translation_date ) {
			return true;
		}
		return false;
```

Reply:
Thank you for the comment, I am also trying to check if the user's first translation was made in at most1 day **before** the event start date. I realised that the `diff()` function returns absolutes regardless if it's before or after the specific date so I needed a way to check that. Therefore, if the date is before the start date, it should return a negative value and if after positive. Perhaps I could change the name of the variable `multiplier`.

Reply:
That's a great explanation, can you add it as a code comment (maybe rewrite it a little to be less prose)?

Reply:
In this situation, I would recommend you to subtract 1 day using the `modify` method, explaining in a comment why we do this, because it is not trivial to understand the reason (I like it).

```suggestion
		// We subtract one day from the first translation date, so we set the translator as a "first time contributor" if she has started translating the previous day to the event starts.
		$first_translation_date = $first_translation_date->modify('-1 day');
		if ( $event_start_date_time <= $first_translation_date ) {
			return true;
		}
		return false;
```

Reply:
@amieiro I like that solution better than a code comment. Nice!

Reply:
@akirk the code comment is only to explain the reason why we subtract one day, because if I see this code without any context, it will be hard to understand for me. That is the only reason to add it, nothing else.

Reply:
That's what I was trying to say: the code change you suggested is the way to go and is better than the explanation that @trymebytes provided.

I think your code comment would be even better if it just explained what we consider as a first time contributor, not what the code does at it speaks for itself.

Something like:
```
// A first time contributor is someone who has never submitted a translation, or only within 1 day before the event.
// Someone who has submitted their first translation before that is not considered a first time contributor.
```

Reply:
Thank you for the idea @amieiro. I just tested and I think it should be written this way;
```suggestion
		$event_start_date_time = $event_start_date_time->modify( '-1 day' );
		if ( $event_start_date_time <= $first_translation_date ) {
		 	return true;
		}
		return false;
```